olive/model/__init__.py

```diff
@@ -749,18 +749,52 @@ def to_json(self, check_object: bool = False):
         return serialize_to_json(config, check_object)
 
 
-class OptimumModel(PyTorchModel):
-    def __init__(self, model_components: List[str], **kwargs):
+class CompositePyTorchModel(PyTorchModel):
+    def __init__(self, model_type: str, model_components: List[Union[str, Dict[str, Any]]], **kwargs):
+        model_file_format = {
+            "Optimum": ModelFileFormat.OPTIMUM,
+            "PyTorchEntireModel": ModelFileFormat.PYTORCH_ENTIRE_MODEL,
+        }[model_type]
+
         kwargs = kwargs or {}
-        kwargs["model_file_format"] = ModelFileFormat.OPTIMUM
-        super().__init__(**kwargs),
-        self.model_components = model_components
+        kwargs["model_file_format"] = model_file_format
+        super().__init__(**kwargs)
+
+        if model_file_format == ModelFileFormat.OPTIMUM:
+            self.model_components = model_components
+        elif model_file_format == ModelFileFormat.PYTORCH_ENTIRE_MODEL:
+            self.model_components = {}
+            self.model_component_names = []
+
+            for model_config in model_components:
+                config_copy = deepcopy(model_config)
+
+                assert "name" in config_copy
+                model_name = config_copy["name"]
+                del config_copy["name"]
+
+                self.model_component_names.append(model_name)
+                self.model_components[model_name] = validate_config(config_copy, ModelConfig).create_model()
+        else:
+            raise NotImplementedError
 
     def to_json(self, check_object: bool = False):
         config = super().to_json(check_object)
         config["config"].update({"model_components": self.model_components})
```

Comment:
I don't think the type of `self.model_components` is correct after the initialization? the original `model_components` was a list but `self.model_components` is a dict for `PYTORCH_ENTIRE_MODEL`.